dotnet/src/SemanticKernel/Memory/Collections/ScoredValue.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.SemanticKernel.Memory.Collections;
```

Comment:
I don't feel great about adding a 'collections' namespace to memory unless these collections are of type "IEmbeddingWithMetadata" or something along those lines. At the same time, since all the new objects are internal, it's not a big deal. I wonder if there's a better place to organize these generic structures though.

Reply:
I agree, usually I like to put things like data structures, extensions for base .NET types etc. to separate project with some abstract name like `DotNetExtensions`, so all projects in solution can reference it and re-use common types. We can discuss it in scope of separate PR, to avoid big expansion of this PR 😄